### PR TITLE
feat: Validate that pathway endpoints have correct types

### PR DESCRIPTION
**Pull request by aababilov**
* a pathway may not lead to a station (location_type=1);
* if a platform has boarding areas, the pathway must be connected to
  its boarding areas instead of the platform itself.

### CODE REVIEW

File: RULES.md [hunk 1]
```diff
@@ -451,8 +453,29 @@ Trip frequencies must not overlap in time
 ##### References:
 * [frequencies.txt specification](http://gtfs.org/reference/static/#frequenciestxt)
 
+<a name="PathwayToPlatformWithBoardingAreasNotice"/>
+
+#### PathwayToPlatformWithBoardingAreasNotice
+
+A pathway has an endpoint that is a platform which has boarding areas. A platform that has boarding
```
Comment (lionel-nj) [inline RULES.md:460]: Here, @isabelle-dr @MobilityData/transit-specs do you think that it's worth updating the specification to clarify this rule (make it more explicit)?

**Review by lionel-nj — CHANGES_REQUESTED** [review]
Thanks @aababilov for your contribution - one comment in-line regarding the naming of `PathwayWrongEndpointTypeNotice`.

**Review by maximearmstrong — APPROVED** [review]
+1 to @lionel-nj comment [here](https://github.com/MobilityData/gtfs-validator/pull/961#discussion_r691211750), but LGTM otherwise!

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/PathwayEndpointTypeValidator.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        noticeContainer.addValidationNotice(
+            new PathwayWrongEndpointTypeNotice(pathway, fieldName, stopId));
+        return;
+      default:
+        break;
+    }
+  }
+
+  /** Describes a pathway which endpoint is a station. */
+  static class PathwayWrongEndpointTypeNotice extends ValidationNotice {
```
[the remark this answers is not in the document]
Comment (aababilov) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/PathwayEndpointTypeValidator.java:88]: This name was chosen on purpose. There is a proposal to add "clusters" as a new location type. Pathways may not be connected to clusters, so this notice name is suitable for both stations and clusters.

[the remark this answers is not in the document]
Comment (aababilov) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/PathwayEndpointTypeValidator.java:88]: But I see your point and let's call it `PathwayToWrongLocationTypeNotice`.

**Pull request merged**